connectors/src/resources/microsoft_resource.ts

```diff
@@ -421,6 +421,29 @@ export class MicrosoftNodeResource extends BaseResource<MicrosoftNodeModel> {
     };
   }
 
+  static async fetchByPaginatedIds({
+    connectorId,
+    pageSize,
+    idCursor,
+  }: {
+    connectorId: ModelId;
+    pageSize: number;
+    idCursor: ModelId;
+  }): Promise<MicrosoftNodeResource[]> {
+    const blobs = await this.model.findAll({
+      where: {
+        connectorId,
+        id: {
+          [Op.gt]: idCursor,
```

Comment:
we already use `nextIdCursor = lastNode.id + 1;` in microsoftNodesGCActivity, so should rather be `Op.gte` ?